src/matlab/homotopy_solver.m

```diff
@@ -39,34 +39,37 @@
     settings_int = varargin{6};
 end
 
-
 import casadi.*
 %%  unfold data
+settings_bkp = settings;
 unfold_struct(settings,'caller')
+settings = settings_bkp;
 unfold_struct(solver_initialization,'caller')
 
 comp_res = model.comp_res;
 nabla_J_fun = model.nabla_J_fun;
 s_elastic_iter = 1;
 
 sigma_k = sigma_0;
-x0 = solver_initialization.lbw(1:model.dimensions.n_x);
+x0 = solver_initialization.w0(1:model.dimensions.n_x);
 
 % lambda00 initialization
 if strcmp(settings.pss_mode, 'Stewart')
-    lambda00 = full(model.lambda00_fun(x0));
+    lambda00 = full(model.lambda00_fun(x0, model.p_global_val));
 elseif strcmp(settings.pss_mode, 'Step')
 %     c_x = full(model.c_fun(x0));
 %     lambda00 = [ max(c_x, 0); min(c_x, 0)];
-    lambda00 = full(model.lambda00_fun(x0));
+    lambda00 = full(model.lambda00_fun(x0, model.p_global_val));
 end
-p_val = [model.p_val(:); lambda00(:)];
+p_val = [model.p_val(:);x0(:); lambda00(:)];
```

Comment:
In Python, the parameters are organized as:
```
p_val = np.concatenate((prob.model.p_val_ctrl_stages.flatten(), np.array([sigma_k, tau_val]), lambda00, x0))
```
i.e. model parameters (given by user), homotopy parameters (introduced when setting up the problem for the homotopy solver), `lambda00, x0` - "eliminated" variables.
It seems that `model.p_val`, gets augmented with the homotopy parameters, which is not ideal IMO, but no "merge blocker" :)

Reply:
Nervermind, I think `model.p_val` only contains the homotopy parameters and you just didnt touch that.